core/commands/swarm.go

```diff
@@ -64,12 +73,32 @@ var swarmPeersCmd = &cmds.Command{
 			return
 		}
 
+		verbose, _, _ := req.Option("verbose").Bool()
 		conns := n.PeerHost.Network().Conns()
 		addrs := make([]string, len(conns))
+		pingService := n.Ping
+
 		for i, c := range conns {
 			pid := c.RemotePeer()
 			addr := c.RemoteMultiaddr()
-			addrs[i] = fmt.Sprintf("%s/ipfs/%s", addr, pid.Pretty())
+
+			if verbose {
+				pContext, cancelFn := context.WithTimeout(req.Context(), timeout)
+				defer cancelFn()
+
+				ch, _ := pingService.Ping(pContext, pid)
```

Comment:
We can grab this information from `n.Peerstore.LatencyEWMA(peer)` no need to ping each peer for it